pandas/core/arrays/string_arrow.py

```diff
@@ -252,9 +255,20 @@ def __len__(self) -> int:
         """
         return len(self._data)
 
-    @classmethod
-    def _from_factorized(cls, values, original):
-        return cls._from_sequence(values)
+    @doc(ExtensionArray.factorize)
+    def factorize(self, na_sentinel: int = -1) -> Tuple[np.ndarray, ExtensionArray]:
+        if self._data.num_chunks == 1:
```

Comment:
Nowadays, `dictionary_encode` works fine for ChunkedArrays as well, so I am not sure this `if` statement is actually needed.

Reply:
Tooks a stab at that in `fletcher` to let CI verify this assumption: Seems to work with `pyarrow` 0.17-2.0 https://github.com/xhochy/fletcher/pull/206